pkg/steps/pre_publish_images.go

```diff
@@ -0,0 +1,189 @@
+package steps
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+
+	imageapi "github.com/openshift/api/image/v1"
+	"github.com/openshift/ci-operator/pkg/api"
+	imageclientset "github.com/openshift/client-go/image/clientset/versioned/typed/image/v1"
+	coreapi "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// PrePublishOutputImageTagStep will ensure that a tag exists
+// in the named ImageStream that resolves to the built
+// pipeline image
+type prePublishOutputImageTagStep struct {
+	config    api.PrePublishOutputImageTagStepConfiguration
+	istClient imageclientset.ImageStreamTagsGetter
+	isClient  imageclientset.ImageStreamsGetter
+	jobSpec   *api.JobSpec
+}
+
+type prepublishConfig struct {
+	namespace, name, tag string
+}
+
+func (s *prePublishOutputImageTagStep) Inputs(ctx context.Context, dry bool) (api.InputDefinition, error) {
+	return nil, nil
+}
+
+func (s *prePublishOutputImageTagStep) getprePublishConfig() (prepublishConfig, error) {
+	if len(s.jobSpec.Refs.Pulls) == 0 {
+		return prepublishConfig{}, fmt.Errorf("did not find a pull request associated with job")
+	}
+	namespace := s.config.To.Namespace
+	if namespace == "" {
+		return prepublishConfig{}, fmt.Errorf("invalid prePublishOutputImageTagStep, namespace must be set")
+	}
+	if len(s.jobSpec.Refs.Pulls) > 1 {
+		return prepublishConfig{}, fmt.Errorf("found more than 1 pull request associated with job, pulls %#v", s.jobSpec.Refs.Pulls)
+	}
+
+	pull := s.jobSpec.Refs.Pulls[0]
+	pullNumber := strconv.Itoa(pull.Number)
+	var name, tag string
+	if s.config.To.Name != "" {
+		name = s.config.To.Name
+		tag = "pr-" + pullNumber
+	} else if s.config.To.Tag != "" {
+		name = "pr-" + pullNumber
+		tag = s.config.To.Tag
+	} else {
+		return prepublishConfig{}, fmt.Errorf("invalid prePublishOutputImageTagStep, one of name or tag must be set")
+	}
+	return prepublishConfig{
+		namespace: namespace,
+		name:      name,
+		tag:       tag,
+	}, nil
+}
+
+func (s *prePublishOutputImageTagStep) Run(ctx context.Context, dry bool) error {
+	conf, err := s.getprePublishConfig()
+	if err != nil {
+		return err
+	}
+
+	log.Printf("Tagging %s into %s/%s:%s", s.config.From, conf.namespace, conf.name, conf.tag)
+
+	fromImage := "dry-fake"
+	if !dry {
+		from, err := s.istClient.ImageStreamTags(s.jobSpec.Namespace).Get(fmt.Sprintf("%s:%s", api.PipelineImageStream, s.config.From), meta.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("could not resolve base image: %v", err)
+		}
+		fromImage = from.Image.Name
+	}
+
+	is := &imageapi.ImageStream{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      conf.name,
+			Namespace: conf.namespace,
+		},
+		// TODO(chance): do I need this?
```

Comment:
No, this just helps people write `image: simple-name` in their PodSpec on the cluster and get auto resolution of image sha